src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -1,478 +1,424 @@
-        this.state.activeApp.suspend({
-          error: function(data, response) {
+  rollbackToAppVersion: function (version) {
+    if (this.state.activeApp != null) {
+      var app = this.state.activeApp;
+      app.setVersion(version);
+      app.save(
+        null,
+        {
+          error: function (data, response) {
             var msg = response.responseJSON.message || response.statusText;
-            alert("Could not suspend: " + msg);
+            alert("Could not update to chosen version: " + msg);
           },
-          success: function() {
+          success: function () {
             // refresh app versions
             this.fetchAppVersions();
           }.bind(this)
         });
-      }
-    },
-
-    poll: function() {
-      this._pollResource();
-    },
-
-    setPollResource: function(func) {
-      // Kill any poll that is in flight to ensure it doesn't fire after having changed
-      // the `_pollResource` function.
-      this.stopPolling();
-      this._pollResource = func;
-      this.startPolling();
-    },
-
-    startPolling: function() {
-      if (this._interval == null) {
-        this.poll();
-        this._interval = setInterval(this.poll, UPDATE_INTERVAL);
-      }
-    },
-
-    stopPolling: function() {
-      if (this._interval != null) {
-        clearInterval(this._interval);
-        this._interval = null;
-      }
-    },
-
-    showAboutModal: function(event) {
-      if (event != null) event.preventDefault();
-
-      if (this.state.modalClass !== null) {
-        return;
-      }
-
-      this.setState({
-        modalClass: AboutModalComponent
-      });
-    },
-
-    showAppModal: function(app) {
-      if (this.state.modalClass !== null) {
-        return;
-      }
-
-      this.setState({
-        activeApp: app,
-        modalClass: AppModalComponent
-      });
-    },
-
-    showNewAppModal: function(event) {
-      if (this.state.modalClass !== null) {
-        return;
-      }
+    }
+  },
 
-      this.setState({
-        modalClass: NewAppModalComponent
-      });
-    },
+  activateTab: function(id) {
+    this.setState({
+      activeTabId: id
+    });
 
-    onTabClick: function(id) {
-      this.setState({
-        activeTabId: id
-      });
+    if (id === tabs[0].id) {
+      this.setPollResource(this.fetchApps);
+    } else if (id === tabs[1].id) {
+      this.setPollResource(this.fetchDeployments);
+    }
+  },
+
+  routeAbout: function () {
+    return (
+      <AboutModalComponent
+        onDestroy={this.modalDestroy}
+        ref="modal" />
+    );
+  },
+
+  routeApps: function(appid, view) {
+    if(!appid || !this.state.activeApp) {
+      return null;
+    }
 
-      if (id === tabs[0].id) {
-        this.setPollResource(this.fetchApps);
-      } else if (id === tabs[1].id) {
-        this.setPollResource(this.fetchDeployments);
-      }
-    },
-
-    render: function() {
-      var modal;
-
-      /* jshint trailing:false, quotmark:false, newcap:false */
```

Comment:
Where did the jshint and jscs comments go? I think there is something missing here